enhancements/security/runtime-class-scc.md

```diff
@@ -0,0 +1,290 @@
+or using a type from the kubernetes API like [the Runtime Class structure][runtime-class-structure],
+`AllowedRuntimeClasses` will be a slice of strings.
+One can think of it as a list of [Handlers][handlers] of the runtime classes that are allowed.
+
+```
+type SecurityContextConstraints struct {
+    ...
+	// RequiredRuntimeClasses is a list of names of runtime classes (also called runtime handlers).
+	// The runtime class specified in a pod request must match one of the items in this list.
+	// To ensure any runtime class, '*' may be specified.
+	// To ensure the default runtime class is used, an empty slice may be specified.
+	// +nullable
+	EnsuredRuntimeClasses []string `json:"ensuredRuntimeClasses" protobuf:"bytes,26,rep,name=ensuredRuntimeClasses"`
+    ...
+}
+```
+Fields in this slice will be logically OR'd.
+A user must specify an item that only matches one field to be permitted to use that runtime class.
+This allows an admin to specify a set of runtime classes that a user is allowed to use.
+
+This gives the admin three options for configuration:
+1. Specifying the empty slice (`[]`) requires pods in the SCC to not request a custom runtime class (meaning it is given the default).
+2. Specifying a set of items (`["openshift-builder"]`) requires pods in the SCC must request a runtime class that matches an item in the list.
+3. Specifying `["*"]` means pods int he SCC can request any runtime class, or omit requesting a custom one.
+
+[runtime-classes]: https://kubernetes.io/docs/concepts/containers/runtime-class/
+[required-drop-capabilities]: https://github.com/openshift/api/blob/0ba2c3658da6cade27a5066575cd8446eb03914c/security/v1/types.go#L53..L55
+[runtime-class-structure]: https://github.com/kubernetes/api/blob/03aa42fe49ac2fa37646cc09e9a84fd825b88117/node/v1beta1/types.go#L37
+[handlers]: https://github.com/kubernetes/api/blob/03aa42fe49ac2fa37646cc09e9a84fd825b88117/node/v1beta1/types.go#L53
+
+### Some Examples
+Below are some RuntimeClasses that will be useful for Openshift to support, as well as reasonining behind not allowing everyone to use them.
+Note, these examples are only meant to demonstrate the current SCC changes fit the range of use cases.
+This proposal does not cover the addition of these runtime classes to openshift.
+- `openshift-builder`: will need access to /dev/fuse on the host, and a user namespace allocated.
+  While these features are available to all users on the host,
+  an untrusted user could take up all available UIDs/GIDs, preventing builds from happening.
+- `openshift-low-latency`: Allows for pods to request the system is configured for latency sensitive workloads.
+  It allows for CPU load balancing, CPU quota disabling, and IRQ smp load balancing to be disabled.
+  A malicious user could get improper access to the host hardware if allowed to use this runtime class.
+- `openshift-sandboxed-containers`: Allows for pods to be run using the kata runtime, as opposed to runc.
```

Comment:
OpenShift Sandboxed Containers would be the product name, but the concrete runtime is kata. So if we are talking handlers, I guess kata would make sense here. There might be a case in the future where sandboxed containers mean more than just kata, then this field would not really be a string but a set instead.

Reply:
updated to `openshift-kata` for now